### PR TITLE
Task#8: Feat - Add filter for manufacturer logos

### LINKED ISSUE #55: Feature: List of phone manufacturers to help users choose
- [x] #56
> As a user, in the app just below the filters, I can see the logos of the mobile phone manufacturers. When I tap on the logos, a list of phones by manufacturer appears directly below so I can choose the phone of the manufacturer I want without having to use the filter above.

### LINKED ISSUE #56: User story#2: Products: Logo filter
- [x] #57

### LINKED ISSUE #57: Task#8 Dev Task - Add filter for manufacturer logos
- [ ] https://github.com/Agility-Internship/Duong-The-Hien-Javascript/issues/55

- Using localStorage.setiTem, using .filter .inclued iterate through the list of products named by brand, then save it to localStorage, attach an event to the getItem button when the user clicks it

### NOTE
- If user want to see all products, user need reload app.
### ACCEPTANCE
- Create list of logos button 

- [x] Under the filter
- [x] Has a rounded edge different from the square ones
- [x] There is a blue border when the user hovers
- [x] Contains the logo's name and trademark

- Implement the  logos button and process the filter results.

- [x] Display a list of phones by manufacturer

-  Testing the features and display of the logos button

- [x] Display phone list when user has submitted to the logo.

### PR DESCRIPTION
**Pull request by TheHienDuong**
- Closes #57
## DEMO

https://github.com/Agility-Internship/Duong-The-Hien-Javascript/assets/113186169/e9dc5252-a3bb-4778-8180-c4f9799ef001

### CODE REVIEW

**Comment by phattruong-agilityio** [comment]
I have some comment on your demo
If you are select brand, the selected brand should be highlight, that help user know which brand is selected

File: practice-02/js/index.js [hunk 1]
```diff
@@ -1,12 +1,25 @@
 import myJson from '../database/products.json' assert {type: 'json'};
 
+/**
+ * This function is used to filter products by name and save them on localStorage
```
Comment (phattruong-agilityio) [inline practice-02/js/index.js]: Why we need to save them on localStorage

File: practice-02/js/index.js [hunk 2]
```diff
@@ -72,3 +85,24 @@ document.addEventListener('DOMContentLoaded', function () {
         });
     });
 });
+
+// Handle click event on logo buttons
+document.addEventListener('DOMContentLoaded', function () {
```
Comment (phattruong-agilityio) [inline practice-02/js/index.js]: We've already have "DOMContentLoaded" event listener

File: practice-02/js/index.js [hunk 3]
```diff
@@ -72,3 +85,24 @@ document.addEventListener('DOMContentLoaded', function () {
...
+    const logoButtons = document.querySelectorAll('.logo-item');
+
+    // TODO: Using localStorage getItem here
```
Comment (phattruong-agilityio) [inline practice-02/js/index.js]: why we need localStorage here

File: practice-02/js/index.js [hunk 4]
```diff
@@ -72,3 +85,24 @@ document.addEventListener('DOMContentLoaded', function () {
...
+    logoButtons.forEach(function (button) {
+        button.addEventListener('click', function () {
+            // Get the name attribute of the clicked logo button's image
+            const name = button.querySelector('img').getAttribute('name');
```
Comment (phattruong-agilityio) [inline practice-02/js/index.js]: what is `name`, too common

File: practice-02/js/index.js [hunk 5]
```diff
@@ -72,3 +85,24 @@ document.addEventListener('DOMContentLoaded', function () {
...
+            filterAndSaveProductsByName(myJson, name);
+
+            // Get data in localStorage and render products
+            const filteredProducts = JSON.parse(localStorage.getItem('filteredProducts'));
```
Comment (phattruong-agilityio) [inline practice-02/js/index.js]: why we need localStorage here

Comment (phattruong-agilityio) [reply practice-02/js/index.js]: IMO, seem our app don't need to use localStorage at all. Please consider about this!

File: practice-02/src/index.html [hunk 6]
```diff
@@ -138,34 +138,34 @@
             <div class="box-logo">
                 <div class="list-logo">
                     <button class="btn logo-item">
-                        <img src="//cdn.tgdd.vn/Brand/1/logo-iphone-220x48.png" width="30" class="logo" alt="Logo iPhone">
+                        <img src="//cdn.tgdd.vn/Brand/1/logo-iphone-220x48.png" width="30" name="iPhone" class="logo" alt="Logo iPhone">
                     </button>
                     <button class="btn logo-item">
-                        <img src="//cdn.tgdd.vn/Brand/1/samsungnew-220x48-1.png" width="30" class="logo" alt="Logo Samsung">
+                        <img src="//cdn.tgdd.vn/Brand/1/samsungnew-220x48-1.png" width="30" name="Samsung" class="logo" alt="Logo Samsung">
```
Comment (phattruong-agilityio) [inline practice-02/src/index.html]: name attr should be lowercase. Update for ALL CASES

File: practice-02/js/index.js (hunk 4 above)
Comment (TheHienDuong) [reply practice-02/js/index.js]: Yes, I changed

File: practice-02/src/index.html (hunk 6 above)
Comment (TheHienDuong) [reply practice-02/src/index.html]: Edited and added .lowCase() method to filter capitalized product names

File: practice-02/js/index.js (hunk 2 above)
Comment (TheHienDuong) [reply practice-02/js/index.js]: Yes, I combined them for the same event

File: practice-02/js/index.js (hunk 3 above)
Comment (TheHienDuong) [reply practice-02/js/index.js]: Yes, local Storage is unnecessary, I removed localStorage

File: practice-02/js/index.js (hunk 1 above)
Comment (TheHienDuong) [reply practice-02/js/index.js]: Yes, local Storage is unnecessary, I removed localStorage

File: practice-02/js/index.js (hunk 2 above)
Comment (TheHienDuong) [reply practice-02/js/index.js]: Yes, I combined them for the same event

File: practice-02/js/index.js (hunk 5 above)
Comment (TheHienDuong) [reply practice-02/js/index.js]: Yes, local Storage is unnecessary, I removed localStorage

**Comment by TheHienDuong** [comment]
Thanks for the feedback on my demo, I resolved the code issues and added highlight when selecting brand. Here is my new demo of the app when the user selects the brand.

#### DEMO

https://github.com/Agility-Internship/Duong-The-Hien-Javascript/assets/113186169/980b5b14-8e5c-4b2f-b94a-b73800c62e53

File: practice-02/js/index.js [hunk 7]
```diff
@@ -71,4 +85,24 @@ document.addEventListener('DOMContentLoaded', function () {
             filterShow.style.display = 'none';
         });
     });
+
+    // Handle click event on logo buttons
+    logoButtons.forEach(function (button) {
+        button.addEventListener('click', function () {
+            const nameProduct = button.querySelector('img').getAttribute('name');
```
Comment (phattruong-agilityio) [inline practice-02/js/index.js]: why get `name` from img, can we get it right on the click button ?

File: practice-02/js/index.js [hunk 8]
```diff
@@ -71,4 +85,24 @@ document.addEventListener('DOMContentLoaded', function () {
...
+            const filteredProducts = filterProductsByName(myJson, nameProduct);
+            renderProductsCard(filteredProducts);
+
+            // Remove the 'selected' class from all logo buttons
+            logoButtons.forEach(function (logoButton) {
```
Comment (phattruong-agilityio) [inline practice-02/js/index.js]: I think I can select multiple products brand, so please make it correct

**Comment by TheHienDuong** [comment]
Thanks for the response on my QUESTION, I've resolved the code issues and updated the multi-select brand feature again. Here is my new demo of the app where the user chooses the brand.

https://github.com/Agility-Internship/Duong-The-Hien-Javascript/assets/113186169/738d27f3-a294-4451-a8b5-b0541941d618

File: practice-02/js/index.js (hunk 7 above)
Comment (TheHienDuong) [reply practice-02/js/index.js]: Yes, I fixed

**Review by phattruong-agilityio — APPROVED** [review]

**Pull request merged**